src/persist-client/src/read.rs

```diff
@@ -76,43 +73,143 @@ pub(crate) enum HollowBatchReaderMetadata<T> {
         as_of: Antichain<T>,
         /// Return all values with time leq `until`.
         until: Antichain<T>,
-        /// After reading the batch, you can downgrade the reader's `since` to
-        /// this value.
-        since: Antichain<T>,
     },
 }
 
+#[derive(Debug)]
+pub(crate) enum LeaseLifeCycle {
+    Issued { seqno: SeqNo, droppable: bool },
+    Consumed { seqno: SeqNo, droppable: bool },
+    Completed,
+}
+
+impl LeaseLifeCycle {
+    fn seqno(&self) -> Option<SeqNo> {
+        use LeaseLifeCycle::*;
+        match self {
+            Issued { seqno, .. } | Consumed { seqno, .. } => Some(*seqno),
+            Completed => None,
+        }
+    }
+}
+
 /// A token representing one read batch.
 ///
 /// This may be exchanged (including over the network). It is tradeable via
-/// [ReadHandle::fetch_batch] for the resulting data stored in the batch.
-#[derive(Debug, Clone, Serialize, Deserialize)]
-#[serde(bound(
-    serialize = "T: Timestamp + Codec64",
-    deserialize = "T: Timestamp + Codec64"
-))]
-#[serde(
-    into = "SerdeReaderEnrichedHollowBatch",
-    from = "SerdeReaderEnrichedHollowBatch"
-)]
-pub struct ReaderEnrichedHollowBatch<T> {
+/// `BatchFetcher::fetch_batch` for the resulting data stored in the batch.
+///
+/// # Panics
+/// `LeasedBatch` panics when dropped unless a very strict set of invariants are
+/// held:
+///
+/// `LeasedBatch` may only be dropped if it:
+/// - Is being converted to `SerdeLeasedBatch`.
+/// - Has been returned to its issuer via `Self::return_lease`.
+///
+/// In any other circumstance, dropping `LeasedBatch` panics.
+#[derive(Debug)]
+pub struct LeasedBatch<T>
```

Comment:
IMO let's push everything `Lease*` into the new fetch mod. they kinda straddle both read and fetch conceptually, but read is already kinda long /shrug